File: app/src/main/java/pl/marczak/mvvmdryrun/base/ValidationContext.java
package pl.marczak.mvvmdryrun.base;


import java.io.InvalidClassException;
import java.util.ArrayList;

public class ValidationContext {
    private ArrayList<Validator> _validators = new ArrayList<>();

    public void addValidator(Validator validator) {
        _validators.add(validator);
    }

    public ArrayList<Validator> getValidators() {
        return _validators;
    }

    public boolean validate() {
        boolean hasValidationErrors = false;
        for (Validator validator : _validators) {
            try {
                boolean validationResult = validator.validate();
                if (!validationResult) {
                    hasValidationErrors = true;
                }
            } catch (InvalidClassException ex) {
                return false;
            }
        }

        return !hasValidationErrors;
    }
}

File: app/src/main/java/pl/marczak/mvvmdryrun/base/validators/ValidationContext.kt
package pl.marczak.mvvmdryrun.base.validators


import java.io.InvalidClassException
import java.util.ArrayList

class ValidationContext {
    val validators = ArrayList<Validator>()

    fun addValidator(validator: Validator) {
        validators.add(validator)
    }

    fun validate(): Boolean {
        var hasValidationErrors = false
        for (validator in validators) {
            try {
                val validationResult = validator.validate()
                if (!validationResult) {
                    hasValidationErrors = true
                }
            } catch (ex: InvalidClassException) {
                return false
            }

        }

        return !hasValidationErrors
    }
}
